game/shieldbattery/shieldbattery.cpp

```diff
@@ -384,13 +386,26 @@ DWORD __stdcall GetModuleFileNameAHook(
   }
 }
 
+HMODULE __stdcall LoadLibraryAHook(_In_ LPCTSTR filename) {
+  if (EndsWith(filename, "local.dll")) {
+    wchar_t starcraft_path[MAX_PATH];
+    GetModuleFileNameW(NULL, starcraft_path, sizeof(starcraft_path));
+    wstring local_dll_path = wstring(starcraft_path);
+    local_dll_path.replace(local_dll_path.end() - strlen("starcraft.exe"), local_dll_path.end(), L"local.dll");
```

Comment:
You can't assume the exe name like this. You also can't assume that the filename won't appear in the filepath leading up to the filename. There is code *in this file* that does this properly. Hint: StartNode